src/routes/user.js

```diff
@@ -5,6 +5,16 @@ const authMiddleware = require('./authMiddleware');
 const logger = require('../logger');
 const deleteUndefinedFields = require('../utils/objectUtils').deleteUndefinedFields;
 
+router.post('/user_exists', async (req, res) => {
+    const username = req.body.username;
+    const user = await userStore.findByUsername(username);
+    if(user) {
+    	res.status(200).json({"exists": true});
+    } else {
+    	res.status(200).json({"exists": false});
+    }
+});
```

Comment:
Is it ok that everyone is able to find which users exist? This is called user enumeration and is usually discouraged. A usual workaround is to just try to authenticate with the username + password that the user gives and see if the authentication was successful.

Reply:
The functionality is required for similar login/registration flow that the current rv uses. This is one of those features that should be put behind additional check such that it cannot be accessed remotely, but only from the rv-terminal itself. #86 

I don't think that user enumeration attack vulnerability is importanyt anyways, as it already exists with the current login as it is prone to timing sidechannel attack. Also with the registration function itself there has to be some kind of check for user existence anyway to avoid duplicate users.